### PR TITLE
Make Query Cache server-level settings reconfigurable at runtime

### PR DESCRIPTION
**Pull request by rschu1ze**
\+ rename setting `max_entry_records` to `max_entry_rows` for consistency with other settings.

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The query cache's server-level settings are now reconfigurable at runtime

### CODE REVIEW

**Comment by yakov-olkhovskiy** [comment]
there is some peculiar clang-tidy finding - didn't know it can do this:
https://s3.amazonaws.com/clickhouse-builds/45758/216636f984eefa4a78eb7ed77519e1dd727aa680/binary_tidy/build_log.log

**Comment by rschu1ze** [comment]
The clang-tidy error looks fixed in master

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -1517,13 +1518,8 @@ try
...
     /// A cache for query results.
-    size_t query_cache_size = config().getUInt64("query_cache.size", 1_GiB);
-    if (query_cache_size)
-        global_context->setQueryCache(
-            query_cache_size,
-            config().getUInt64("query_cache.max_entries", 1024),
-            config().getUInt64("query_cache.max_entry_size", 1_MiB),
-            config().getUInt64("query_cache.max_entry_records", 30'000'000));
+    global_context->setQueryCache();
+    global_context->updateQueryCacheConfiguration(config());
```
Comment (yakov-olkhovskiy) [inline programs/server/Server.cpp:1522]: don't you think `setQueryCache` is loosing its meaning here? maybe make it to accept config parameter right away?

**Review by yakov-olkhovskiy — APPROVED** [review]
overall LGTM, but consider my comment above

**Pull request merged**